src/main/java/org/jenkinsci/plugins/tuleap_git_branch_source/TuleapSCMSource.java

```diff
@@ -86,14 +88,45 @@ public class TuleapSCMSource extends AbstractGitSCMSource {
     private String credentialsId;
     private TuleapAccessToken credentials;
 
-    @DataBoundConstructor
     public TuleapSCMSource(TuleapProject project, TuleapGitRepository repository) {
         this.repository = repository;
         this.project = project;
         this.projectId = String.valueOf(project.getId());
         this.repositoryPath = repository.getPath();
     }
 
+    @DataBoundConstructor
+    public TuleapSCMSource(String projectId, String repositoryPath) {
```

Comment:
If it is possible, I think it would be better if this constructor also populated `this.project` and `this.projectId` itself so that we are always sure of what we have.

Reply:
I already thought to do that, but I think this is not a good idea because of the `DataBoundConstructor`, it does some automagic binding, and it may introduce some unwanted side effect which can be hard to debug

Reply:
Isn't the `@DataBounConstructor` only taking into account the arguments of the constructior ie `projectId` and ` repositoryPath`  ? 

https://javadoc.jenkins.io/component/stapler/org/kohsuke/stapler/DataBoundConstructor.html

> The matching is done by using the constructor parameter name

Or are your worried about something else that I don't see ? I haven't looked into this in quite a while...